src/main/java/Duke.java

```diff
@@ -1,10 +1,312 @@
+                    }
+                }
+                else if (input.length() >= 7 && input.substring(0,7).toUpperCase().equals(Commands.DELETE.getString())) {
+                    try {
+                        int index = Integer.parseInt(input.substring(7).trim());
+                        deleteTask(index);
+                    } catch (NumberFormatException ex) {
+                        printError("Please input an Integer for the \"Delete\" command.");
+                    } catch (DukeException ex) {
+                        printError(ex.getMessage());
+                    }
+                }
+                else if (input.length() >= 5 && input.substring(0,5).toUpperCase().equals(Commands.TODO.getString())) {
+                    try {
+                        String name = input.substring(5).trim();
+                        addToStorageList(Todo.createTodo(name));
+                    } catch (DukeException ex) {
+                        printError(ex.getMessage());
+                    }
+                }
+                else if (input.length() >= 6 && input.substring(0,6).toUpperCase().equals(Commands.EVENT.getString())) {
+                    try {
+                        int limiterPosition = input.indexOf(" /at ");
+                        String name;
+                        String timing;
+                        if (limiterPosition != -1) {
+                            name = input.substring(6, limiterPosition).trim();
+                            timing = input.substring(limiterPosition + 5).trim();
+                        } else {
+                            throw new DukeException("Missing date for Event task");
+                        }
+                        addToStorageList(Event.createEvent(name, timing));
+                    } catch (DukeException ex) {
+                        printError(ex.getMessage());
+                    }
+                }
+                else if (input.length() >= 9 && input.substring(0,9).toUpperCase().equals(Commands.DEADLINE.getString())) {
+                    try {
+                        int limiterPosition = input.indexOf(" /by ");
+                        String name;
+                        String dueDate;
+
+                        if (limiterPosition != -1) {
+                            name = input.substring(9, limiterPosition).trim();
+                            dueDate = input.substring(limiterPosition + 5).trim();
+                        } else {
+                            throw new DukeException("Missing deadline for Deadline task");
+                        }
+                        addToStorageList(Deadline.createDeadline(name, dueDate));
+                    } catch (DukeException ex) {
+                        printError(ex.getMessage());
+                    }
+                }
+                else {
+                    printError("Sorry I don't know what that means :(");
+                }
+            }
+        }
```

Comment:
Perhaps you can try segment this code since it is a bit too long now imo, maybe you can consider an `execute()` for each command?